mlflow/models/utils.py

```diff
@@ -858,6 +882,109 @@ def _is_scalar(x):
         return _enforce_unnamed_col_schema(pf_input, input_schema)
 
 
+def _convert_pf_to_dict(pdf: pd.DataFrame):
+    if not isinstance(pdf, pd.DataFrame):
+        raise MlflowException(f"Expected input to be DataFrame. Found: {type(pdf).__name__}")
+    # We expect the pandas DataFrame to be a single row
+    # with columns for each key in the dictionary
+    # dict like {index -> {column -> value}}
```

Comment:
We can use code here:

```
pd.DataFrame(...)
```

This is less ambiguous and easier to understand.

Reply:
I'm going to refactor this based on our discussion this morning :D

Reply:
Sounds good :)